main.py

```diff
@@ -131,73 +145,147 @@ def calculate_interest(self, rate: float) -> None:
         print("Debit accounts do not earn interest.")
 
 
-class Customer:
-    def __init__(self, account_number: str, account: Account) -> None:
+class SavingsAccount:
+    def __init__(self, account_number: str, balance: float = 0) -> None:
         self.account_number = account_number
-        self.account = account
+        self.balance = balance
+        self.is_locked = False
+        self.transactions = []
+
+    def deposit(self, amount: float) -> None:
+        if not self.is_locked and amount > 0:
+            self.balance += amount
+            self.transactions.append(Transaction("Deposit", amount, datetime.datetime.now()))
+            print(f"Deposited ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid deposit amount.")
+
+    def withdrawal(self, amount: float) -> None:
+        if self.is_locked:
+            self.balance -= amount
+            self.transactions.append(Transaction("Withdrawal", amount, datetime.datetime.now()))
+            print(f"Withdrew ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid withdrawal amount or insufficient funds.")
+
+    def get_balance(self):
+        return self.balance
+
+    def lock_account(self) -> None:
```

Comment:
Decide whether you explicitly define that you return `None` always or do not do it at all.